Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -82,31 +103,54 @@ sub initPlugin {
 
 sub shutdownPlugin {
 	my @played = values %recentlyPlayed;
+
 	$prefs->set('recent', \@played);
 }
 
+sub updateRecentlyPlayed {
+	my ($class, $client, $song) = @_;
+	my ($url) = unwrap($song->originUrl);
+
+	$recentlyPlayed{$url} ||= { 
+			url      => $url,
+			title    => $song->track->title,
+			cover    => Slim::Player::ProtocolHandlers->iconForURL($url, $client),
+			duration => $song->duration,
+	};	
+}
+
+sub unwrap {
+	return shift =~ m|^podcast://([^&]+)(?:&from=(\d+)$)?|;	
+}
+
+sub wrap {
+	my ($url, $from) = @_;
```

Comment:
Please call these `wrapUrl` or something instead of just wrap.

Reply:
done